### PR TITLE
Updated Chapter 12.3.1 

### PR DESCRIPTION
**Pull request by OkoliEvans**
The chapter did not explicitly explain how user defined structs are written to Storage, I found this might be confusing for newbies so I added some clarifications.
@enitrat

### CODE REVIEW

**Comment by enitrat** [comment]
> The chapter did not explicitly explain how user defined structs are written to Storage

Yes and it's still on purpose, it's not just user defined structs it's all structs, even u256 values, which are structs, are written in the same way to storage.

What I am opposed to is the _specialization_ of the explanation specifically for structs. For all types stored, there is one single common way of reading and writing them, it's not specific to structs. The only specificity for structs is that you need to derive starknet::Store on them

File: src/ch99-01-03-01-storage-variables.md [hunk 1]
```diff
@@ -30,6 +30,21 @@ The base storage address for structs remains `sn_keccak(variable_name)`, and the
 
 Subsequent fields are stored in addresses contiguous to the first elements at addresses `base_address + i`.
 
+
+## Writing user defined structs to storage
```
Comment (enitrat) [inline src/ch99-01-03-01-storage-variables.md:34]: Therefore you can remove this explanation as it doesn't make sense to have this for structs explicitly.

File: src/ch99-01-03-01-storage-variables.md [hunk 2]
```diff
@@ -30,6 +30,21 @@ The base storage address for structs remains `sn_keccak(variable_name)`, and the
...
+
+To write user defined struct to storage you have to first add the struct variables to the default Storage struct. The variable type is the user defined struct identifier.
```
Comment (enitrat) [inline src/ch99-01-03-01-storage-variables.md:36]: It's not just structs, it's every single storage variable!

**Review by enitrat — CHANGES_REQUESTED** [review]

File: src/ch99-01-03-01-storage-variables.md (hunk 2 above)
Comment (OkoliEvans) [reply src/ch99-01-03-01-storage-variables.md:36]: Okay, I get it now

**Comment by OkoliEvans** [comment]
I made the changes, but I'm not sure the wordings are perfect. If there's need to still edit the wordings, some suggestions will really help... like a sample text with the right words / structure. @enitrat

**Comment by enitrat** [comment]
Here are some leads on how to proceed:

- Do not mention storing user-defined structs _at all_. As there is no special treatment for user-defined structs compared to other types in storage, we will not explicitly talk about how they're stored.
- Replace all integrations of code in this page by specific corresponding snippets from `listings/ch99-starknet-smart-contracts/listing_99_03_example_contract`. This ensure that the explanations are in line with the example introduced in the previous chapter.
- If you believe that it's not clear that variables must be added to the storage struct to be considered in storage, you can eventually reword this part. But I do think it's clear enough 🤔 

```rust
Storage variables in Starknet contracts are stored in a special struct called `Storage`:

```rust, noplayground
{{#rustdoc_include ../listings/ch99-starknet-smart-contracts/listing_99_02/src/lib.cairo:here}}
```

<span class="caption">Listing 99-2: A Storage Struct</span>

The storage struct is a [struct](./ch05-00-using-structs-to-structure-related-data.md) like any other,
except that it **must** be annotated with `#[storage]` allowing you to store mappings using the `LegacyMap` type.
```

**Comment by OkoliEvans** [comment]
New changes made @enitrat

**Comment by enitrat** [comment]
I will close this as I don't want to spend too much time on back-and-fourths review. 
I opened another PR with the changes that I would have liked to see in this section, if you wanted to improve the clarity and organization #427

For #419, if you still intend to do it, you can add your part under the `### Structs storage layout`, named `### Enums storage layout`

**Comment by OkoliEvans** [comment]
I will continue on the Enums then.

**Pull request closed without merging**